### PR TITLE
ci: Travis: check logs for TSan also  [skip appveyor]

### PR DESCRIPTION
**Pull request by blueyed**
Ref: https://github.com/neovim/neovim/pull/10591#issuecomment-521248233

### CODE REVIEW

File: ci/common/build.sh [hunk 1]
```diff
@@ -85,11 +85,11 @@ build_nvim() {
 
   # Invoke nvim to trigger *San early.
   if ! (bin/nvim --version && bin/nvim -u NONE -e -c ':qall') ; then
-    asan_check "${LOG_DIR}"
-    exit 1
+    ret=$?
+    sanitizer_check "${LOG_DIR}"
+    exit $?
```
Comment (blueyed) [inline ci/common/build.sh]: This is meant to exit with 66 in case of TSan failures (for clarity).

**Comment by blueyed** [comment]
Better failure now: https://travis-ci.org/neovim/neovim/builds/571897736#L788

File: ci/common/test.sh [hunk 2]
```diff
@@ -80,8 +80,8 @@ valgrind_check() {
   check_logs "${1}" "valgrind-*"
 }
 
-asan_check() {
-  if test "${CLANG_SANITIZER}" = "ASAN_UBSAN" ; then
+sanitizer_check() {
```
Comment (justinmk) [inline ci/common/test.sh]: since we're renaming it let's put the verb first, similar to `check_logs`.
```suggestion
check_sanitizer() {
```

**Comment by justinmk** [comment]
Nice, thank you! The `starting` data race in `tui_grid_resize` is known/expected, but nice to see that TSan catches it now. The check is "best effort", i.e. we don't really care about the race, so I hope there's a way to tell TSan to ignore it.

**Comment by blueyed** [comment]
Suppressions can be used: https://github.com/google/sanitizers/wiki/ThreadSanitizerCppManual#suppressing-reports

**Comment by justinmk** [comment]
> Suppressions can be used: https://github.com/google/sanitizers/wiki/ThreadSanitizerCppManual#suppressing-reports

yeah, we have `src/.asan-blacklist` for ASan.

**Review by justinmk — APPROVED** [review]

**Comment by blueyed** [comment]
> yeah, we have src/.asan-blacklist for ASan.

Hmm.. maybe we should have added it to allowed-failures here, until it is addressed?

**Comment by justinmk** [comment]
can make a `src/.tsan-blacklist`

File: ci/common/build.sh [hunk 3]
```diff
@@ -84,12 +84,11 @@ build_nvim() {
   fi
 
   # Invoke nvim to trigger *San early.
-  if ! (bin/nvim --version && bin/nvim -u NONE -e -c ':qall') ; then
-    asan_check "${LOG_DIR}"
+  if ! (bin/nvim --version && bin/nvim -u NONE -e -cq | cat -A) ; then
```
Comment (blueyed) [inline ci/common/build.sh]: `cat -A` does not work on osx.. fix in https://github.com/neovim/neovim/pull/10779

**Pull request merged**